File: VB/CustomDrawButtonEvent/CustomButtonEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Registrator
Imports System.ComponentModel

Namespace CustomDrawButtonEvent
	Public Delegate Sub DrawButtonEventHandler(ByVal sender As Object, ByVal e As DrawButtonEventArgs)

	<UserRepositoryItem("Register")> _
	Public Class MyButtonEdit
		Inherits ButtonEdit

		Shared Sub New()
			RepositoryItemMyButtonEdit.Register()
		End Sub

		Public Sub New()
		End Sub


		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemMyButtonEdit.EditorName
			End Get
		End Property


		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemMyButtonEdit
			Get
				Return TryCast(MyBase.Properties, RepositoryItemMyButtonEdit)
			End Get
		End Property


		<Description("Occurs when custom editor buttons are being drawn."), Category(CategoryName.Events)> _
		Public Custom Event DrawButton As DrawButtonEventHandler
			AddHandler(ByVal value As DrawButtonEventHandler)
				AddHandler Properties.DrawButton, value
			End AddHandler
			RemoveHandler(ByVal value As DrawButtonEventHandler)
				RemoveHandler Properties.DrawButton, value
			End RemoveHandler
			RaiseEvent(ByVal sender As Object, ByVal e As DrawButtonEventArgs)
			End RaiseEvent
		End Event


	End Class
End Namespace

File: CS/CustomDrawButtonEvent/CustomButtonEdit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Registrator;
using System.ComponentModel;

namespace CustomDrawButtonEvent {
    public delegate void DrawButtonEventHandler(object sender, DrawButtonEventArgs e);

    [UserRepositoryItem("Register")]
    public class MyButtonEdit : ButtonEdit {

        static MyButtonEdit() {
            RepositoryItemMyButtonEdit.Register();
        }

        public MyButtonEdit() {
        }


        public override string EditorTypeName {
            get { return RepositoryItemMyButtonEdit.EditorName; }
        }


        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemMyButtonEdit Properties {
            get { return base.Properties as RepositoryItemMyButtonEdit; }
        }


        [Description("Occurs when custom editor buttons are being drawn."), Category(CategoryName.Events)]
        public event DrawButtonEventHandler DrawButton {
            add { Properties.DrawButton += value; }
            remove { Properties.DrawButton -= value; }
        }

        
    }
}
